test/ClangImporter/foreign_errors.swift

```diff
@@ -108,3 +113,24 @@ func testNSErrorExhaustive() {
     }
   }
 }
+
+func testBadOverrides(obj: FoolishErrorSub) throws {
+  try obj.performRiskyOperation()
+  let _: FoolishErrorSub = try obj.produceRiskyOutput()
+  let _: String = try obj.produceRiskyString()
+
+  let _: NSObject = try obj.badNullResult()
+  let _: CInt = try obj.badNullResult2() // This is unfortunate but consistent.
```

Comment:
The original code is wrong. Maybe Clang should be diagnosing the bogus override (an `int` instead of an object pointer? seriously?), but I think our generated IR is reasonable.